include/common_structs.h

```diff
@@ -2283,4 +2283,19 @@ typedef struct ActionCommandStatus {
     /* 0x74 */ s16 mashMeterCutoffs[6]; // upper bounds for each interval
     /* 0x80 */ s8 mashMeterIntervals;
 } ActionCommandStatus;
+
+struct PopupMessage;
+typedef void (*PopupMessageCallback)(struct PopupMessage* popup);
+typedef struct PopupMessage {
```

Comment:
Assuming it works, I think it would better to do:
```suggestion
typedef struct PopupMessage PopupMessage;
typedef void (*PopupMesageCallback)(PopupMessage* popup);
struct PopupMessage { /* ... */ };
```